components/PlayerButtons/PlayerButtons.tsx

```diff
@@ -0,0 +1,55 @@
+import { Button } from '@/components/ui/button';
+import { usePlayer } from '@/providers/audio/PlayerProvider';
+import { useFeedProvider } from '@/providers/FeedProvider';
+import { TrackMetadata } from '@/types/Track';
+import Image from 'next/image';
+import { useEffect } from 'react';
+import { MdPauseCircle, MdPlayCircle } from 'react-icons/md';
+
+type PlayerButtonProps = {
+  metadata?: TrackMetadata;
+};
+
+export default function PlayerButtons({ metadata }: PlayerButtonProps) {
+  const [player, dispatch] = usePlayer();
+  const { handleNext, handlePrev } = useFeedProvider();
+
+  useEffect(() => {
+    if (player.position !== 0 && player.position > player.duration - 1000) {
+      handleNext();
+    }
+  }, [player.position]);
+
+  if (!metadata) return <></>;
```

Comment:
please avoid returning empty components.

in this case, you can `conditionally render` the `<PlayerButtons />` in the parent for only when `metadata` is present, right?